gen/safeq

```diff
@@ -141,25 +136,32 @@ perunServicesInit::finalize;
 ##############################################################################
 
 sub processUsers {
-	my $memberAttributes = shift;
-	my $resourceAttributes = shift;
-	my $login = $memberAttributes->{$A_USER_LOGIN};
-
-	unless(defined $usersByLogin->{$memberAttributes->{$A_USER_LOGIN}}) {
-	my %tmp =  ( %$memberAttributes, %$resourceAttributes );
-		$usersByLogin->{$memberAttributes->{$A_USER_LOGIN}} = $memberAttributes;
-		$usersByLogin->{$memberAttributes->{$A_USER_LOGIN}}->{$A_R_PREPAID} = $resourceAttributes->{$A_R_PREPAID};
-		$usersByLogin->{$memberAttributes->{$A_USER_LOGIN}}->{$A_R_PRINT_FOR_FREE} = ( $resourceAttributes->{$A_R_PRINT_FOR_FREE} || 0) ;
+	my $memberId = shift;
+	my $resourceId = shift;
+	my $login = $data->getUserAttributeValue(attrName => $A_USER_LOGIN, member => $memberId);
+
+	unless(defined $usersByLogin->{$login}) {
+		$usersByLogin->{$login}->{$A_USER_FIRST_NAME} = $data->getUserAttributeValue(attrName => $A_USER_FIRST_NAME, member => $memberId);
+		$usersByLogin->{$login}->{$A_USER_LAST_NAME} = $data->getUserAttributeValue(attrName => $A_USER_LAST_NAME, member => $memberId);
+		$usersByLogin->{$login}->{$A_USER_MAIL} = $data->getUserAttributeValue(attrName => $A_USER_MAIL, member => $memberId);
+		$usersByLogin->{$login}->{$A_USER_LOGIN} = $data->getUserAttributeValue(attrName => $A_USER_LOGIN, member => $memberId);
```

Comment:
I do not think you need to store the login again. You can use the `$login` key during the Printing phase.

Reply:
You are right, I updated it.